conjure/src/main/java/com/palantir/conjure/cli/ConjureCli.java

```diff
@@ -43,20 +51,70 @@
             .setSerializationInclusion(JsonInclude.Include.NON_ABSENT);
 
     public static void main(String[] args) {
-        System.exit(new CommandLine(new ConjureCli()).execute(args));
+        System.exit(prepareCommand().execute(args));
+    }
+
+    @VisibleForTesting
+    static CommandLine prepareCommand() {
+        return new CommandLine(new ConjureCli()).setExecutionExceptionHandler(new ExceptionHandler());
+    }
+
+    public static final class ExceptionHandler implements IExecutionExceptionHandler {
+        @Override
+        public int handleExecutionException(Exception ex, CommandLine commandLine, ParseResult _parseResult)
+                throws Exception {
+            if (ex == null) {
+                return 0;
+            }
+            // For known internal errors, continue, as they're signaling errors on the input, not in the code
+            Throwable rootCause = Throwables.getRootCause(ex);
+            if (!(rootCause instanceof ConjureException)
+                    && !(rootCause instanceof ParseException)
+                    && !(rootCause instanceof CyclicImportException)) {
+                throw ex;
+            }
+            if (!(commandLine.getCommand() instanceof ConjureCliCommand)) {
+                throw ex;
+            }
+            ConjureCliCommand cmd = commandLine.getCommand();
+            if (cmd.isVerbose()) {
+                // If the command is verbose, print the full stack trace
+                throw ex;
+            }
+
+            // Dedupe exceptions to ensure we don't end up in a cause loop
+            Set<Throwable> seen = new HashSet<>();
```

Comment:
nit:
```suggestion
            Set<Throwable> seen = Collections.newSetFromMap(new IdentityHashMap<>());
```

Reply:
Ah, I see. I guess exceptions kind of fall outside of normal object equality rules a bit? Ended up refactoring this code away though.